engines/order_management/spec/services/order_management/order/updater_spec.rb

```diff
@@ -128,6 +128,46 @@ module Order
           expect(updater).not_to receive(:update_shipments).with(order)
           updater.update
         end
+
+        context "with pending payments" do
+          let!(:payment) { create(:payment, order: , amount: order.total) }
+
+          context "with order in payment state" do
+            let(:order) { create(:order_with_totals, state: "payment") }
+
+            it "updates pending payments" do
+              # update order so the order total will change
+              update_order_quantity(order)
+              order.payments.reload
+
+              expect { updater.update }.to change { payment.reload.amount }.from(10).to(20)
+            end
+          end
+
+          context "with order in confirmation state" do
+            let(:order) { create(:order_with_totals, state: "confirmation") }
+
+            it "updates pending payments" do
+              # update order so the order total will change
+              update_order_quantity(order)
+              order.payments.reload
+
+              expect { updater.update }.to change { payment.reload.amount }.from(10).to(20)
+            end
+
+          end
+
+          context "with order in cart" do
+            let(:order) { create(:order_with_totals) }
+
+            it "doesn't update pending payments" do
+              # update order so the order total will change
+              update_order_quantity(order)
+
+              expect { updater.update }.not_to change { payment.reload.amount }
```

Comment:
I was just testing existing logic so I don't know the reasoning behind this. 
From memory, if the order is the cart, it shouldn't have pending payment yet has the customer would not have gone through the payment step. That said, there might be scenarios where this could happen I am not sure. Anyway, payment will be updated has the customer goes through "payment" and "confirmation" step.

Reply:
When the checkout fails, we reset the order to cart state: `OrderCheckoutRestart`